### PR TITLE
[Merged by Bors] - chore(Radical): minor refactoring of lemmas

### PR DESCRIPTION
**Pull request by seewoo5**
---

Seperated from #18882 

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/RingTheory/Radical.lean [hunk 1]
```diff
@@ -83,14 +83,14 @@ theorem radical_eq_of_associated {a b : M} (h : Associated a b) : radical a = ra
   unfold radical
   rw [h.primeFactors_eq]
 
-theorem radical_unit_eq_one {a : M} (h : IsUnit a) : radical a = 1 :=
+theorem radical_of_isUnit {a : M} (h : IsUnit a) : radical a = 1 :=
   (radical_eq_of_associated (associated_one_iff_isUnit.mpr h)).trans radical_one_eq
 
-theorem radical_unit_mul {u : Mˣ} {a : M} : radical ((↑u : M) * a) = radical a :=
-  radical_eq_of_associated (associated_unit_mul_left _ _ u.isUnit)
+theorem radical_isUnit_mul_left {a u : M} (h : IsUnit u) : radical (u * a) = radical a :=
```
Comment (YaelDillies) [inline Mathlib/RingTheory/Radical.lean:89]:
```suggestion
theorem radical_mul_of_isUnit_left {a u : M} (h : IsUnit u) : radical (u * a) = radical a :=
```
will be more discoverable

**Review by YaelDillies — APPROVED** [review]
Thanks!

maintainer delegate

**Review by jcommelin — COMMENTED** [review]
bors d+

**Pull request closed without merging**